app/api/events/getEvents.ts

```diff
@@ -0,0 +1,69 @@
+const formatEvent = (event) => {
+  const url = event.url && fixRelativeUrl(event.url);
+  return {
+    Presenter: event.presenterName,
+    Url: {
+      Description: event.title, // this is alt text for the link, ask matt what he wants to do for it
+      Url: url,
+    },
+    title: event.title,
+    Thumbnail: {
+      Url: fixRelativeUrl(event.thumbnail),
+      Description: event.thumbnailDescription,
+    },
+    PresenterProfileUrl:
+      event.presenterProfileUrl && fixRelativeUrl(event.presenterProfileUrl),
+    StartDateTime: event.startDateTime,
+    EndDateTime: event.endDateTime,
+    StartShowBannerDateTime: event.startShowBannerDateTime,
+    EndShowBannerDateTime: event.endShowBannerDateTime,
+    LiveStreamUrl: event.youTubeId
+      ? `https://www.youtube.com/watch?v=${event.youTubeId}`
+      : null,
+    CalendarType: event.calendarType,
+    City: event.city,
+    Enabled: event.enabled,
+    EventShortDescription: descriptionToPlainText(event.description),
+    ShowOnPowerSessions: "No",
+    NoteInternal: event.internalNote,
+    HasVideo: event.youTubeId ? "Yes" : "No",
+    YouTubeId: event.youTubeId,
+    PresenterDescription:
+      event.presenterList && event.presenterListlength > 0
+        ? event.presenterList[0].presenter.about
+        : null,
+  };
+};
+
+export const getEventsWithClient = async (eventClient, presenterName) => {
+  const events = [];
+  // TODO: remove client side after fixing events with multiple presenters in the name
+  for (const event of eventClient.data.eventsCalendarConnection.edges) {
+    if (
+      event.node.presenterName &&
+      event.node.presenterName.includes(presenterName)
+    )
+      events.push(formatEvent(event.node));
+    if (events.length === 10) {
+      break;
+    }
+  }
+  return events;
+};
+
+const fixRelativeUrl = (url: string) => {
+  if (url.startsWith("/")) {
+    return `https://ssw.com.au${url}`;
+  }
+  return url;
+};
+
+const descriptionToPlainText = (description) => {
```

Comment:
it looks like this is formatting html and not plain text

Reply:
I was trying to emulate the format the original API was using for this field, which would use divs an linebreaks to separate paragraphs. I figured I could achieve the same effect using <p> tags to create line breaks (by default Tina stores line breaks as separate "paragraph" props in the json file, I'm just converting it back to html).
![image](https://github.com/user-attachments/assets/fbe6dedf-dc5c-4374-82f0-9aa17af4117f)

Reply:
I was commenting on the name of the function since it doesn't do what it says on the box

Ideally, I'd avoid adding extra logic and just fix the data at the source - this can be done as a separate pbi

Reply:
Ok, I'll rename the function but I think fixing the function is a low priority since I can't see the prop being used anywhere on rules. We might even just be able to omit it from the API.